File: Cantidad de dígitos de un número/ANSI C/cuenta_digitos_v5.c
/* Creado por Andrés Melgar
 * para iterando++ http://iterando.online/
 * entrada: http://iterando.online/ejercicios-resueltos/cantidad-de-digitos-de-un-numero-en-c/
 */
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int cuenta_digitos(int numero);

int main() {
    int numero;
    printf("Ingrese un número: ");
    scanf("%d", &numero);

    numero = abs(numero);
    printf("El número tiene %d dígito(s)\n", cuenta_digitos(numero));
    return 0;
}

int cuenta_digitos(int numero) {
    if (numero == 0)
        return 1;
    return log10(numero) + 1;
}

File: Cantidad de dígitos de un número/Java/Cuenta_Digitos_v5.java
package cuenta_digitos_v5;

import static java.lang.Math.abs;
import static java.lang.Math.log10;
import java.util.Scanner;

public class Cuenta_Digitos_v5 {

    public static void main(String[] args) {
        int numero;
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese un número: ");
        numero = reader.nextInt();

        numero = abs(numero);
        System.out.printf("El número tiene %d dígito(s)%n", cuenta_digitos(numero));
    }

    private static int cuenta_digitos(int numero) {
        if (numero == 0) {
            return 1;
        }
        return (int)log10(numero) + 1;
    }

}
